core/common/buffer.hpp

```diff
@@ -35,12 +35,10 @@ namespace fc::common {
      */
     Buffer(size_t size, uint8_t byte);
 
-    ~Buffer() = default;
-
     /**
      * @brief lvalue construct buffer from a byte vector
      */
-    Buffer(std::vector<uint8_t> &&v);
+    explicit Buffer(std::vector<uint8_t> &&v);
```

Comment:
move-constructor may be implicit, because it doesn't copy, and accepts only rvalue which is not used elsewhere